### PR TITLE
Changes tags in StatsContext proto from string to bytes.

### PR DESCRIPTION
**Pull request by dinooliva**
And documents encoding format.

### CODE REVIEW

File: stats/stats_context.proto [hunk 1]
```diff
@@ -16,6 +16,17 @@ package google.instrumentation;
...
 option java_outer_classname = "StatsContextProto";
 
+// Wire format of instrumentation stats (aka Census) tags.
 message StatsContext {
-  string tags = 1;
+  // Encodes all stats tags (aka Census tags) to propagate over an RPC in a
+  // single byte vector.
+  //
+  // The format is:
+  //  num_tags [key_len key_bytes value_len value_bytes]*
```
Comment (a-veitch) [inline stats/stats_context.proto]: As we talked about IRL, I don't think we need the num_tags. But lets talk about this some more.

File: stats/stats_context.proto [hunk 2]
```diff
@@ -16,6 +16,17 @@ package google.instrumentation;
...
+  //
+  // All implementations must honor these specs:
+  //  * The num_tags is represented using varint.
+  //  * The key/value lengths are represented using varint.
+  //  * keys/values do not include the terminating character '\0'.
```
Comment (a-veitch) [inline stats/stats_context.proto]: I don't think we need this condition. In fact, I think we should state that tag keys/values must be in UTF-8 format (another possibility is US-ASCII, but I think we will have cases where people want UTF-8, especially for values).

File: stats/stats_context.proto [hunk 3]
```diff
@@ -16,6 +16,17 @@ package google.instrumentation;
...
 message StatsContext {
-  string tags = 1;
+  // Encodes all stats tags (aka Census tags) to propagate over an RPC in a
+  // single byte vector.
+  //
+  // The format is:
+  //  num_tags [key_len key_bytes value_len value_bytes]*
+  //
+  // All implementations must honor these specs:
+  //  * The num_tags is represented using varint.
```
Comment (a-veitch) [inline stats/stats_context.proto]: Should provide a pointer to varint documentation, e.g. https://developers.google.com/protocol-buffers/docs/encoding#varints

Comment (dinooliva) [inline stats/stats_context.proto]: Done.

File: stats/stats_context.proto (hunk 2 above)
Comment (dinooliva) [inline stats/stats_context.proto]: We last agreed upon US-ASCII, so I've added that to the requirements. We should review this in our next inter-language sync.

File: stats/stats_context.proto (hunk 1 above)
Comment (dinooliva) [inline stats/stats_context.proto]: Ack.

Comment (dinooliva) [inline stats/stats_context.proto]: If we want num_tags, maybe it should be a different field in the proto.

**Pull request merged**